### ISSUE TITLE
MeanWeightedSumQuantileLoss and ND metrics produce incorrect results for all-zero data

### ISSUE THREAD

**Issue by shchur** [awslabs/gluon-ts#3034]
## Description
In the new evaluation module `gluonts.ev`, the `MeanWeightedSumQuantileLoss` and `ND` metrics are computed incorrectly if the target time series equals all zeros during the forecast horizon.

## To Reproduce

```python
import numpy as np
import pandas as pd
from gluonts.model.forecast import SampleForecast
from gluonts.model.evaluation import evaluate_forecasts
from gluonts.dataset.split import split
from gluonts.ev.metrics import MeanWeightedSumQuantileLoss, ND

T = 50
H = 20
start = pd.Period("2020-01-01", freq="D")
data = [{"target": np.zeros(T + H), "start": start}]
_, test_template = split(dataset=data, offset=-H)
test_data = test_template.generate_instances(H, windows=1)
forecasts = [SampleForecast(np.ones([1, H]), start_date=start + T)]
evaluate_forecasts(forecasts, test_data=test_data, metrics=[MeanWeightedSumQuantileLoss([0.1, 0.5, 0.9]), ND()])
```

## Error message or code output
Code snippet above produces the following output:

```
      mean_weighted_sum_quantile_loss  ND[0.5]
None                              0.0      NaN
```

Expected output:
```
      mean_weighted_sum_quantile_loss  ND[0.5]
None                              Inf      Inf
```

The `NaN` value for `ND` is arguably okay, but the 0.0 value of WQL is definitely wrong.

Old evaluator from `gluonts.evaluation` used to produce `Inf` values in both of these cases.

## Environment
- Operating system: Amazon Linux 2
- Python version: 3.10
- GluonTS version: 0.14.0

**Comment by lostella** [comment]
I think `NaN` should be the expected value in both cases, why would `Inf` be ok?

**Comment by shchur** [comment]
Numpy and pandas return `Inf` when dividing a positive float by zero and `NaN` when dividing zero by zero. As far as I understand, that's what happens for WQL and ND metrics when the denominator equals zero.
```python
>>> np.array(2.0) / 0.0
inf
>>> np.array(0.0) / 0.0
nan
>>> (pd.Series([2.0]) / 0.0).item()
inf
>>> (pd.Series([0.0]) / 0.0).item()
nan
```

**Comment by lostella** [comment]
Right, I missed that the prediction is ones, not zeros, OK

**Comment by lostella** [comment]
A more minimal working example:

```python
import numpy as np
from gluonts.ev.metrics import MeanWeightedSumQuantileLoss, ND

H = 20

data = {
    "label": np.zeros((H)),
    "0.5": np.ones(H),
    "0.1": np.ones(H),
    "0.9": np.ones(H),
}

print(ND()().update(data).get())
print(MeanWeightedSumQuantileLoss([0.1, 0.5, 0.9])().update(data).get())
```

Yields

```python
-- # masked
0.0
```

**Comment by lostella** [comment]
Fixed in #3037
